src/features/processPicker.ts

```diff
@@ -65,17 +67,27 @@ export class RemoteAttachPicker {
             return Promise.reject<string>(new Error("Configuration \"" + name + "\" in launch.json does not have a " +
                 "pipeTransport argument with debuggerPath for pickRemoteProcess. Use pickProcess for local attach."));
         } else {
-            let pipeProgram = args.pipeTransport.pipeProgram;
-            let pipeArgs = args.pipeTransport.pipeArgs;
-            let platformSpecificPipeTransportOptions = RemoteAttachPicker.getPlatformSpecificPipeTransportOptions(args);
+            let pipeProgram: string = args.pipeTransport.pipeProgram;
+            let pipeArgs: string[] = args.pipeTransport.pipeArgs;
+            let quoteArgs: boolean = args.pipeTransport.quoteArgs != null ? args.pipeTransport.quoteArgs : true; // default value is true
+            let platformSpecificPipeTransportOptions: any = RemoteAttachPicker.getPlatformSpecificPipeTransportOptions(args);
 
             if (platformSpecificPipeTransportOptions) {
                 pipeProgram = platformSpecificPipeTransportOptions.pipeProgram || pipeProgram;
                 pipeArgs = platformSpecificPipeTransportOptions.pipeArgs || pipeArgs;
+                quoteArgs = platformSpecificPipeTransportOptions.pipeTransport.quoteArgs != null ? platformSpecificPipeTransportOptions.pipeTransport.quoteArgs : quoteArgs;
             }
 
-            let argList = RemoteAttachPicker.createArgumentList(pipeArgs);
-            let pipeCmd: string = `"${pipeProgram}" ${argList}`;
+            let pipeCmdList: string[] = [];
+            pipeCmdList.push(pipeProgram);
+            pipeCmdList = pipeCmdList.concat(pipeArgs);
+
+            const scriptShellCmdList: string[] = ["bash", "-s"];
```

Comment:
>bash [](start = 51, length = 4)

I am not sure it really matters much for .NET, but should we use 'sh' instead of 'bash' to make it easier to port code between this extension and C++?